lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorState.java

```diff
@@ -22,17 +22,150 @@
 import org.apache.reef.annotations.audience.Private;
 
 /**
- * Various states that the EvaluatorManager could be in. The EvaluatorManager is
- * created when a resource has been allocated by the ResourceManager.
+ * Various states that the EvaluatorManager could be in.
+ * The EvaluatorManager is created when a resource has been allocated by the ResourceManager.
  */
 @DriverSide
 @Private
 enum EvaluatorState {
-  ALLOCATED,  // initial state
-  SUBMITTED,  // client called AllocatedEvaluator.submitTask() and we're waiting for first contact
-  RUNNING,    // first contact received, all communication channels established, Evaluator sent to client.
-  CLOSING,    // evaluator is asked shutdown, but not closed yet.
-  DONE,       // clean shutdown
-  FAILED,     // some failure occurred.
-  KILLED      // unclean shutdown
+
+  /** Initial state. */
+  ALLOCATED,
+
+  /** Client called AllocatedEvaluator.submitTask() and we're waiting for first contact. */
+  SUBMITTED,
+
+  /** First contact received, all communication channels established, Evaluator sent to client. */
+  RUNNING,
+
+  /** Evaluator is asked to shut down, but has not closed yet. */
+  CLOSING,
+
+  /** Clean shutdown. */
+  DONE,
+
+  /** Some failure occurred. */
+  FAILED,
+
+  /** Unclean shutdown. */
+  KILLED;
+
+  /**
+   * Check if evaluator is in the initial state (ALLOCATED).
+   * @return true if allocated, false otherwise.
```

Comment:
Should `allocated` here be uppercase, as name of state? Same question for `submitted` in next function